Fixes to wall_setter for bugs when num_cars == 1

Kaiyotech opened the pull request.

Kaiyotech committed:
Fixes to allow working when num_cars == 1

Rolv-Arild commented:
Swapping teams like this kind of defeats the point, for example if you have a goalie setter it would put an attacker where the defender should be.

Kaiyotech commented:
Sorry, I probably fixed stuff that I didn't test. I'll take another look
and update it. I mostly intended to just fix it so it didn't crash if you
only had one car.

Thanks.

On Wed, Apr 13, 2022, 7:05 AM Rolv-Arild ***@***.***> wrote:

> Swapping teams like this kind of defeats the point, for example if you
> have a goalie setter it would put an attacker where the defender should be.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/RLGym/rlgym-tools/pull/32#issuecomment-1097918651>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AWLB4KXKCGZ5T7C6QJTUAVTVE2S7LANCNFSM5SOGZMCQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

Kaiyotech commented:
Yeah, sorry, I didn't even realize I had added that commit to the PR. I removed that commit and you can just do the other one. I didn't end up using the swap teams anyway, I ended up swapping stuff inside my setter, for the exact reason you mentioned. It just seemed weird that the function was there with no way to call it, so I added it with a default of False so it matched the existing behavior, but it doesn't really matter.

Rolv-Arild committed:
Merge branch 'main' into my_changes
